ADAL/src/ADAuthenticationResult+Internal.m

```diff
@@ -216,6 +216,18 @@ + (ADAuthenticationResult *)resultFromBrokerResponse:(NSDictionary *)response
     ADTokenCacheItem* item = [ADTokenCacheItem new];
     [item setAccessTokenType:@"Bearer"];
     BOOL isMRRT = [item fillItemWithResponse:response];
+    
+    // A bug in previous versions of broker would override the provided authority in some cases with
+    // common. If the intended tenant was something other then common then the access token may
+    // be bad, so clear it out. We will force a token refresh later.
+    NSArray *pathComponents = [[NSURL URLWithString:item.authority] pathComponents];
+    NSString *tenant = pathComponents.count > 1 ? pathComponents[1] : nil;
+    BOOL fValidTenant = response[@"vt"] != nil || [tenant isEqualToString:@"common"];
+    if (!fValidTenant)
+    {
+        item.accessToken = nil;
+    }
+    
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the useless trailing whitespaces at the end of this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)